From 0914b5ebb1f67287e2bb7bfa2d6d620aabc07463 Mon Sep 17 00:00:00 2001
From: Alvaro Karsz <alvaro.karsz@solid-run.com>
Date: Thu, 25 May 2023 13:15:54 +0300
Subject: [PATCH 14/45] Support reading MAC address for ravb driver

Implement the read_rom_hwaddr callback for the ravb network driver.
Read the MAC address from EEPROM.

Signed-off-by: Alvaro Karsz <alvaro.karsz@solid-run.com>
---
 drivers/net/ravb.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/ravb.c b/drivers/net/ravb.c
index 5a5a242d..0a4cd2bf 100644
--- a/drivers/net/ravb.c
+++ b/drivers/net/ravb.c
@@ -369,6 +369,11 @@ static int ravb_write_hwaddr(struct udevice *dev)
 	return 0;
 }
 
+static int ravb_read_hwaddr(struct udevice *dev)
+{
+	return mac_read_from_eeprom();
+}
+
 /* E-MAC init function */
 static int ravb_mac_init(struct ravb_priv *eth)
 {
@@ -701,6 +706,7 @@ static const struct eth_ops ravb_ops = {
 	.free_pkt		= ravb_free_pkt,
 	.stop			= ravb_stop,
 	.write_hwaddr		= ravb_write_hwaddr,
+	.read_rom_hwaddr	= ravb_read_hwaddr,
 };
 
 int ravb_of_to_plat(struct udevice *dev)
-- 
2.41.0

